specification/healthbot/Healthbot.Management/models.tsp

```diff
@@ -0,0 +1,460 @@
+
+  /**
+   * KeyVault properties for the resource encryption.
+   */
+  keyVaultProperties?: KeyVaultProperties;
+
+  /**
+   * The access control method for the Azure Health Bot resource.
+   */
+  @visibility(Lifecycle.Read)
+  accessControlMethod?: string;
+}
+
+/**
+ * Properties of the key vault.
+ */
+model KeyVaultProperties {
+  /**
+   * The name of the key vault key.
+   */
+  keyName: string;
+
+  /**
+   * The version of the key vault key.
+   */
+  keyVersion?: string;
+
+  /**
+   * The Uri of the key vault.
+   */
+  keyVaultUri: string;
+
+  /**
+   * The user assigned identity (ARM resource id) that has access to the key.
+   */
+  userIdentity?: string;
+}
+
+/**
+ * The resource model definition for a ARM tracked top level resource
+ */
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "For backward compatibility"
+model TrackedResource extends Resource {
+  /**
+   * Resource tags.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-no-record" "For backward compatibility"
+  @visibility(Lifecycle.Read, Lifecycle.Create, Lifecycle.Update)
+  tags?: Record<string>;
+
+  /**
+   * The geo-location where the resource lives
+   */
+  @visibility(Lifecycle.Read, Lifecycle.Create)
+  location: string;
+}
+
+/**
+ * The resource model definition for a ARM tracked top level resource
+ */
+model Resource {
+  /**
+   * Fully qualified resource Id for the resource.
+   */
+  @visibility(Lifecycle.Read)
+  id?: string;
+
+  /**
+   * The name of the resource
+   */
+  @visibility(Lifecycle.Read)
+  name?: string;
+
+  /**
+   * The type of the resource.
+   */
+  @visibility(Lifecycle.Read)
+  type?: string;
+
+  /**
+   * Metadata pertaining to creation and last modification of the resource
+   */
+  @visibility(Lifecycle.Read)
+  systemData?: SystemData;
+}
+
+/**
+ * Read only system data
+ */
+model SystemData {
+  /**
+   * The identity that created the resource.
+   */
+  createdBy?: string;
+
+  /**
+   * The type of identity that created the resource
+   */
+  createdByType?: IdentityType;
+
+  /**
+   * The timestamp of resource creation (UTC)
+   */
+  // FIXME: (utcDateTime) Please double check that this is the correct type for your scenario.
+  createdAt?: utcDateTime;
+
+  /**
+   * The identity that last modified the resource.
+   */
+  lastModifiedBy?: string;
+
+  /**
+   * The type of identity that last modified the resource
+   */
+  lastModifiedByType?: IdentityType;
+
+  /**
+   * The timestamp of resource last modification (UTC)
+   */
+  // FIXME: (utcDateTime) Please double check that this is the correct type for your scenario.
+  lastModifiedAt?: utcDateTime;
+}
+
+/**
+ * The resource management error response.
+ */
+@error
+model Error {
+  /**
+   * The error object.
+   */
+  #suppress "@azure-tools/typespec-client-generator-core/property-name-conflict" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  error?: {
```

Comment:
Can we directly give a name for this anonymous model as m4 name? Here would be ErrorError?

Since we agreed we would report errors in TCGC side if any anonymous models are disallowed?